great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -126,27 +126,32 @@ def _convert_root_connection_detail_fields(cls, values: dict) -> dict:
         for field_name in tuple(values.keys()):
             if field_name in connection_detail_fields:
                 connection_details[field_name] = values.pop(field_name)
+
+        if values.get("connection_string") and connection_details:
+            raise ValueError(
+                "Cannot provide both a connection string or a combination of account, user, and password."
+            )
+
         if connection_details:
             values["connection_string"] = connection_details
         return values
 
-    @pydantic.root_validator
-    def _check_xor_input_args(cls, values: dict) -> dict:
+    @pydantic.validator("connection_string")
+    def _check_connection_string(
```

Comment:
Renamed since it wasn't actually doing XOR check and was just validating connection_string.